### PR TITLE
(MAINT) Allow Vanagon to follow HTTP redirects

### PR DESCRIPTION
**Pull request by mckern**
Otherwise we're liable to get a lot of 404s based on something as
trivial as character case.

### CODE REVIEW

File: lib/vanagon/component/source/http.rb [hunk 1]
```diff
@@ -48,24 +48,28 @@ def verify
         # Downloads the file from @url into the @workdir
         #
```
Comment (haus) [inline lib/vanagon/component/source/http.rb]: YARD some params?

Comment (mckern) [inline lib/vanagon/component/source/http.rb]: I YARDed the param.

**Comment by mckern** [comment]
This is sort of hilarious

```ruby
>> uri = URI.parse("https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz")
=> #<URI::HTTPS:0x007f88de82dbb0 URL:https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz>
>> download(uri)
Downloading file 'cfpropertylist-2.2.7.tar.gz' from url 'https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz'
Downloading file 'cfpropertylist-2.2.7' from url 'https://codeload.github.com/ckruse/CFPropertyList/tar.gz/cfpropertylist-2.2.7'
=> "cfpropertylist-2.2.7.tar.gz"
>>
```

```shell
mckern@flexo vanagon (git:maint/follow_redirects) $ ls cfpropertylist-2.2.7*
cfpropertylist-2.2.7
mckern@flexo vanagon (git:maint/follow_redirects) $ 
```

**Comment by mckern** [comment]
Fixed

```ruby
>> uri = URI.parse("https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz")
=> #<URI::HTTPS:0x007fd4b32034f8 URL:https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz>
>> download(uri)
Downloading file 'cfpropertylist-2.2.7.tar.gz' from url 'https://github.com/ckruse/CFPropertyList/archive/cfpropertylist-2.2.7.tar.gz'
Downloading file 'cfpropertylist-2.2.7.tar.gz' from url 'https://codeload.github.com/ckruse/CFPropertyList/tar.gz/cfpropertylist-2.2.7'
=> "cfpropertylist-2.2.7.tar.gz"
```

```bash
mckern@flexo vanagon (git:maint/follow_redirects) $ ls ./cfpropertylist-2.2.7*
./cfpropertylist-2.2.7.tar.gz
mckern@flexo vanagon (git:maint/follow_redirects) $
```

**Comment by shrug** [comment]
👍

**Pull request merged**